### ISSUE TITLE
no-reassign flags wrong side of object destructuring

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#26]
```javascript
import { key } from './module'
const { key: path } = this.props.row
```

erronously flags `key` as a re-assign, while conversely
```javascript
import { key } from './module'
const { path: key } = this.props.row
```

is not reported (but is the true re-assignment),
